src/nvim/testdir/test_highlight.vim

```diff
@@ -761,4 +761,137 @@ func Test_highlight_default_colorscheme_restores_links()
   hi clear
 endfunc
 
+" Test for the hlget() function
+func Test_hlget()
+  call assert_notequal([], filter(hlget(), 'v:val.name == "Visual"'))
+  call assert_equal([], hlget('SomeHLGroup'))
+  highlight MyHLGroup cterm=reverse ctermfg=10 ctermbg=Black
+  call assert_equal([{'id': hlID('MyHLGroup'), 'ctermfg': '10', 'name': 'MyHLGroup', 'ctermbg': '0', 'cterm': {'reverse': v:true}}], hlget('MyHLGroup'))
+  highlight clear MyHLGroup
+  call assert_equal(v:true, hlget('MyHLGroup')[0].cleared)
+  highlight link MyHLGroup IncSearch
+  call assert_equal('IncSearch', hlget('MyHLGroup')[0].linksto)
+  highlight clear MyHLGroup
+  " call assert_equal([], hlget(v:null))
```

Comment:
This fails?

Reply:
Yeah it fails with v:null . Didn't know about v:_null_string v:_null_dict . Those will probably fix this . I was wondering how to get a null with string type.